### PR TITLE
TGUI vending machines (picked up from PR #9419)

### PR DESCRIPTION
**Pull request by ruben-svensson**
[QOL] [TGUI]
## About the PR 
(Picking up from Skeletonman0's cool PR #9419, made some changes but could still use a re-review)

this PR aims to convert the existing html UI for vending machines into a slightly nicer looking TGUI interface.
player vending machines work with this as well

the menu for the wires and player vending machine settings is collapsable

176808980-3d456fd0-e208-426d-8c1c-79ab8fe5feaf

## Why's this needed? 

TGUI good, older interfaces that use raw html and chui bad

```changelog
(u)Skeletonman0 & Grokberg
(+)Converted the UI of vending machines to TGUI

### CODE REVIEW

**Comment by ruben-svensson** [comment]
Removed old HTML, code, currently going through the changes that were requested from the old pr.

File: code/modules/vending/vending.dm [hunk 1]
```diff
@@ -45,7 +58,8 @@
 	mats = 20
 	layer = OBJ_LAYER - 0.1 // so items get spawned at 3, don't @ me
 	deconstruct_flags = DECON_SCREWDRIVER | DECON_WRENCH | DECON_MULTITOOL
-	object_flags = CAN_REPROGRAM_ACCESS | NO_GHOSTCRITTER
+	object_flags = CAN_REPROGRAM_ACCESS
```
Comment (ZeWaka) [inline code/modules/vending/vending.dm:61]: why ghostcritter?

Comment (ruben-svensson) [reply code/modules/vending/vending.dm:61]: I would like @Skeletonman0 's reason on this, not sure why this is the case

File: code/modules/vending/vending.dm [hunk 2]
```diff
@@ -35,6 +36,18 @@
 		src.product_hidden = hidden
 		src.logged_on_vend = logged_on_vend
 
+	proc/getBase64Img()
+		var/path = src.product_path
+		. = product_base64_cache[path]
+		if(isnull(.))
+			var/atom/dummy_atom = new path // people demand overlays on their vending machine bottles
+			SPAWN(0)
```
[the remark this answers is not in the document]
Comment (ruben-svensson) [reply code/modules/vending/vending.dm:44]: I would like @Skeletonman0 's reason on this, I don't really know what purpose this part provides

[the remark this answers is not in the document]
Comment (Skeletonman0) [reply code/modules/vending/vending.dm:44]: iirc i copy pasted most of the base64 image stuff from somewhere else. i think i stole it from the plantmaster and modified it. probably doesnt need the spawn(0)

File: code/modules/vending/vending.dm (hunk 1 above)
Comment (Skeletonman0) [reply code/modules/vending/vending.dm:61]: i dont remember changing that, probably should have the NO_GHOSTCRITTER again

Comment (ruben-svensson) [reply code/modules/vending/vending.dm:61]: Alright, I'll change it back.

File: code/modules/vending/vending.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (ruben-svensson) [reply code/modules/vending/vending.dm:44]: I'll remove it and test it.

[the remark this answers is not in the document]
Comment (ruben-svensson) [reply code/modules/vending/vending.dm]: Worked!

**Comment by ZeWaka** [comment]
> I noticed that since this change, the sound effect for an item being vended doesn't sound the same. Instead of what sounds like a knob being twisted, *then* an item hitting the bottom where it can be grabbed... Now it plays both at the same time.
> 
> Hopefully that can be fixed, but in any case: the new vending UI looks so much better than the old one!

@LikeLakers2 if you could file a bug report for that, that'd be great

**Pull request merged**